compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -359,10 +359,8 @@ impl<'tcx> InferCtxt<'tcx> {
     pub fn get_impl_future_output_ty(&self, ty: Ty<'tcx>) -> Option<Ty<'tcx>> {
         let (def_id, substs) = match *ty.kind() {
             ty::Alias(_, ty::AliasTy { def_id, substs, .. })
-                if matches!(
-                    self.tcx.def_kind(def_id),
-                    DefKind::OpaqueTy | DefKind::ImplTraitPlaceholder
-                ) =>
+                if matches!(self.tcx.def_kind(def_id), DefKind::OpaqueTy)
```

Comment:
Similarly, could you split this into two branches `ty::Alias(ty::Opaque)` and `ty::Alias(_) if is_impl_trait_in_trait`?